libexec/pyenv

```diff
@@ -72,11 +73,13 @@ fi
 
 if [ -z "${PYENV_DIR}" ]; then
   PYENV_DIR="$PWD"
-else
-  cd "$PYENV_DIR" 2>/dev/null || abort "cannot change working directory to \`$PYENV_DIR'"
-  PYENV_DIR="$PWD"
-  cd "$OLDPWD"
 fi
+
+if [ ! -d "$PYENV_DIR" ] || [ ! -e "$PYENV_DIR" ]; then
+  abort "cannot change working directory to \`$PYENV_DIR'"
+fi
+
+PYENV_DIR=$(cd "$PYENV_DIR" && echo "$PWD")
```

Comment:
Just in case, please add quoting around the `$(...)`.

```sh
PYENV_DIR="$(cd "$PYENV_DIR" && echo "$PWD")"
```

Reply:
Might not be necessary altogether: https://github.com/pyenv/pyenv/pull/1036#discussion_r151895707

Reply:
I should have followed the discussion little more.... Anyway it's good enough to me to proceed for merging~